extern/cfe/src/bmi_cfe.c

```diff
@@ -956,26 +926,42 @@ static int Get_var_itemsize (Bmi *self, const char *name, int * size)
 }
 
 
-// TODO: complete implementation
-static int Get_var_location(Bmi *self, const char *name, char *location)
+static int Get_var_location (Bmi *self, const char *name, char * location)
 {
+    // Check to see if in output array first
+    for (int i = 0; i < OUTPUT_VAR_NAME_COUNT; i++) {
+        if (strcmp(name, output_var_names[i]) == 0) {
+            strncpy(location, output_var_locations[i], BMI_MAX_UNITS_NAME);
+            return BMI_SUCCESS;
+        }
+    }
+    // Then check to see if in input array
```

Comment:
Any particular reason to check output then input?  Could the variable possibly be in both?

Reply:
Variables defined only as output.

Reply:
This looks like it is templated from my original logic for `Get_var_units`.  It seemed reasonable for that to potentially  provide unit information for both input and output variables.  I assume the same is true for location information

There aren't currently input variables defined, but there were at one point and very easily could be again (I can think of at least one specific change worth considering that would require adding them).  I.e.,  pre-writing logic to account for there being a collection of input variables, and just having the collection be empty for now, seems sensible.